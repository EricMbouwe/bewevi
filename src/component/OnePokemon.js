import PropTypes from 'prop-types';
import Logo from './Logo';
// import EmptyImage from './EmptyImage';

const OnePokemon = ({
  pokemon, handleError,
}) => (
  <div className="one-pokemon main">
    <div className="one-pokemon logo">
      <Logo />
    </div>
    <div className="one-pokemon top-part">

      <div className="one-pokemon main-image">
        <img
          src={`https://raw.githubusercontent.com/PokeAPI/sprites/master/sprites/pokemon/other/official-artwork/${pokemon.id}.png`}
          alt="pokemon"
          onError={handleError}
          className="one-pokemon top-images"
        />
      </div>
      <div className="one-pokemon top-description">
        <div>
          Name:
          {' '}
          {pokemon.name}
        </div>

        {pokemon.weight && (
          <div>
            Weight:
            {' '}
            {pokemon.height * 0.1}
            kg
          </div>
        )}

        {pokemon.height && (
          <div>
            Height:
            {' '}
            {pokemon.height * 10}
            cm
          </div>
        )}
      </div>
    </div>
    <div className="one-pokemon bottom-part">
      <div className="one-pokemon one-property">
        <div>
          Fruit
        </div>

      </div>

      <div className="one-pokemon one-property">
        <div>
          Leaf
        </div>

      </div>
      <div className="one-pokemon one-property">
        <div>
          Flower
        </div>

      </div>
    </div>

  </div>

);

OnePokemon.propTypes = {
  pokemon: PropTypes.shape().isRequired,
  handleError: PropTypes.func.isRequired,
};

export default OnePokemon;
