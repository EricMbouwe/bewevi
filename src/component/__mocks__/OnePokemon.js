import PropTypes from 'prop-types';

const OnePokemon = ({
  pokemon,
  handleError,
}) => (
  <>
    {JSON.stringify(pokemon)}
    <img src="#" alt="test" onError={handleError} />
  </>
);

OnePokemon.propTypes = {
  pokemon: PropTypes.shape().isRequired,
  handleError: PropTypes.func.isRequired,
};

export default OnePokemon;
