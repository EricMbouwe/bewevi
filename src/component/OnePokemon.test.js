import React from 'react';
import { render, screen } from '@testing-library/react';
import '@testing-library/jest-dom';
import renderer from 'react-test-renderer';
import ReactTestUtils from 'react-dom/test-utils';
import { BrowserRouter } from 'react-router-dom';
import OnePokemon from './OnePokemon';

const handleError = jest.fn();
const pokemon = {
  name: 'venusaur',
  weight: 0.2,
  height: 20,

};
const color = 'green';
const habitat = 'grassland';
const shape = 'quadruped';

const evolutionChain = ['1', '2', '3'];
describe('<OnePokemon />', () => {
  it('contains expected texts', () => {
    render(
      <BrowserRouter>
        <OnePokemon
          pokemon={pokemon}
          handleError={handleError}
          color={color}
          habitat={habitat}
          shape={shape}
          evolutionChain={evolutionChain}
        />
      </BrowserRouter>,
    );
    expect(screen.getByText(/grassland/i)).toBeInTheDocument();
    expect(screen.getByText(/quadruped/i)).toBeInTheDocument();
    expect(screen.getByText(/2kg/i)).toBeInTheDocument();
    expect(screen.getByText(/200cm/i)).toBeInTheDocument();
  });

  it('triggers onError when there are errors of img tags', () => {
    const rendered = ReactTestUtils.renderIntoDocument(
      <BrowserRouter>
        <OnePokemon
          pokemon={pokemon}
          handleError={handleError}
          color={color}
          habitat={habitat}
          shape={shape}
          evolutionChain={evolutionChain}
        />
      </BrowserRouter>,
    );

    const imgs = ReactTestUtils.scryRenderedDOMComponentsWithTag(rendered, 'img');
    imgs.map(img => ReactTestUtils.Simulate.error(img));
    expect(handleError.mock.calls.length).toBe(4);
  });

  it('renders correctly', () => {
    const tree = renderer.create(
      <BrowserRouter>
        <OnePokemon
          pokemon={pokemon}
          handleError={handleError}
          color={color}
          habitat={habitat}
          shape={shape}
          evolutionChain={evolutionChain}
        />
      </BrowserRouter>,
    ).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
