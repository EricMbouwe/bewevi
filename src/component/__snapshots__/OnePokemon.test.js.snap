// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<OnePokemon /> renders correctly 1`] = `
<div
  className="one-pokemon main"
>
  <div
    className="one-pokemon logo"
  >
    <a
      className="logo main"
      href="/"
      onClick={[Function]}
    >
      <img
        alt="logo"
        className="logo image"
        src="emptyImage.svg"
      />
      <div
        className="logo title"
      >
        vy
      </div>
    </a>
  </div>
  <div
    className="one-pokemon top-part"
  >
    <div
      className="one-pokemon main-image"
    >
      <img
        alt="pokemon"
        className="one-pokemon top-images"
        onError={[MockFunction]}
        src="www.ozovalihasan.com"
      />
    </div>
    <div
      className="one-pokemon top-description"
    >
      <div
        className="one-pokemon description-part"
      >
        Year:
          
      </div>
      <div
        className="one-pokemon description-part"
      >
        Common Name:
          
        No common name
      </div>
      <div
        className="one-pokemon description-part"
      >
        Scientific Name:
          
        No scientific name
      </div>
      <div
        className="one-pokemon description-part"
      >
        Common Family Name:
          
        No common family name
      </div>
      <div
        className="one-pokemon description-part"
      >
        Family Name:
          
        No family name
      </div>
    </div>
  </div>
  <div
    className="one-pokemon bottom-part"
  >
    <div
      className="one-pokemon one-property"
    >
      <div>
        Fruit
      </div>
      <img
        alt="fruit of undefined"
        className="one-pokemon bottom-images"
        onError={[MockFunction]}
      />
    </div>
    <div
      className="one-pokemon one-property"
    >
      <div>
        Leaf
      </div>
      <img
        alt="Leaf of undefined"
        className="one-pokemon bottom-images"
        onError={[MockFunction]}
      />
    </div>
    <div
      className="one-pokemon one-property"
    >
      <div>
        Flower
      </div>
      <img
        alt="Flower of undefined"
        className="one-pokemon bottom-images"
        onError={[MockFunction]}
      />
    </div>
  </div>
</div>
`;
